hudi-io/src/main/java/org/apache/hudi/io/hfile/HFileFileInfoBlock.java

```diff
@@ -61,4 +69,39 @@ public HFileInfo readFileInfo() throws IOException {
     }
     return new HFileInfo(fileInfoMap);
   }
+
+  // ================ Below are for Write ================
+
+  public HFileFileInfoBlock(HFileContext context) {
+    super(context, HFileBlockType.FILE_INFO, -1L);
+  }
+
+  public void add(String name, byte[] value) {
+    entries.put(name, value);
+  }
+
+  @Override
+  public ByteBuffer getPayload() {
+    ByteBuffer buff = ByteBuffer.allocate(context.getBlockSize() * 2);
+    HFileProtos.InfoProto.Builder builder =
+        HFileProtos.InfoProto.newBuilder();
+    for (Map.Entry<String, byte[]> e : entries.entrySet()) {
+      HFileProtos.BytesBytesPair bbp = HFileProtos.BytesBytesPair
+          .newBuilder()
+          .setFirst(ByteString.copyFrom(getUTF8Bytes(e.getKey())))
+          .setSecond(ByteString.copyFrom(e.getValue()))
```

Comment:
Could we call `wrap` instead of `copy`?
```suggestion
          .setFirst(ByteString.wrap(getUTF8Bytes(e.getKey())))
          .setSecond(ByteString.wrap(e.getValue()))
```

Reply:
Cannot; wrap is a package-private fucntion.

Reply:
Got it, I'm thinking to avoid copy overhead.  Not a big issue though.  Let's keep it as is.